src/ConstructiveSolidGeometry/SurfacePrimitives/ToroidalAnnulus.jl

```diff
@@ -97,4 +97,13 @@ function sample(t::ToroidalAnnulus{T}, g::CartesianTicksTuple{T})::Vector{Cartes
         if (r_tubeMin <= hypot(r - t.r_torus, z - t.z) <= r_tubeMax) &&
            (t.θ === nothing || _in_angular_interval_closed(mod(atan(z - t.z, r - t.r_torus), T(2π)), t.θ))
     ]
-end
\ No newline at end of file
+end
+
+function distance_to_surface(point::AbstractCoordinatePoint{T}, t::ToroidalAnnulus{T})::T where {T}
+    pcy = CylindricalPoint(point)
+    Δφ = pcy.φ - t.φ
+    z, r_on_plane = pcy.r .* sincos(Δφ)
+    pp = PlanarPoint{T}(r_on_plane, pcy.z) - PlanarVector{T}(t.r_torus, t.z)
+    pct_transformed = CartesianPoint{T}(pp.u, pp.v, z)
+    distance_to_surface(pct_transformed, CylindricalAnnulus(T, t.r_tube, t.θ, T(0)))
+end
```

Comment:
Am I understanding it right that you perform a coordinate transformation here, such that the `ToroidalAnnulus` is in the `x`-`y`-plane?

Reply:
Yes!